src/registries/base-registry.js

```diff
@@ -99,9 +99,10 @@ export default class BaseRegistry {
     });
   }
 
-  async init(): Promise<void> {
+  async init(overrides: Object = {}): Promise<void> {
     this.mergeEnv('yarn_');
     await this.loadConfig();
+    Object.assign(this.config, overrides);
```

Comment:
This overrides the `registry` defined in `.yarnrc` with `undefined` if a `--registry` flag is not passed.

More specifically, the code that calls this will pass `{registry: undefined}` if `--registry` is not passed.

```
Object.assign({registry: 'something_from_yarnrc'}, {registry: undefined});
>returns: {registry: undefined}
```

Reply:
😮 my bad for missing it!
thanks for the quick response @BYK